kindelia_core/src/runtime/mod.rs

```diff
@@ -1631,7 +1678,28 @@ impl Runtime {
             let cont = ask_arg(self, term, 1);
             let normalized = self.normalize(unnormalized, mana, &mut HashSet::new())?;
             let cont = alloc_app(self, cont, normalized);
-            let done = self.run_io(subject, caller, cont, mana);
+            let done = self.run_io(subject, caller, cont, mana, call_term, events);
+            clear(self, host, 1);
+            clear(self, term.get_loc(0), 2);
+            return done;
+          }
+          IO_EMIT => {
+            let term_to_emit = ask_arg(self, term, 0);
+            // checks if args is a number
+            if self.check_num(term_to_emit, mana).is_err() {
+              // if not a number, checks if is a flat CTR
+              self.check_ctr_argm(subject, None, mana, term_to_emit)?;
+            }
+            if let Some(call_term) = call_term {
+              events.push(Event {
+                fnid: subject,
+                argm: call_term,
```

Comment:
I'm also not sure how storing `RawTerm`s is working here. I'm pretty sure it should be `readback`'ed to a `Term` as in the `done_term` field.

Reply:
I avoided using `readback` or doing any transforms on `run_io` itself. So I send the RawCell's back as a result, and in the `run_statement` function, after checking everything and doing the readback of the `done_term` is that I actually read what is written in these `RawCell`'s (I used the `show_term`, but was going to ask you if I should use `readback` instead).